code/modules/projectiles/guns/ballistic.dm

```diff
@@ -506,6 +550,60 @@ GLOBAL_LIST_INIT(gun_saw_types, typecacheof(list(
 		update_icon()
 		return TRUE
 
+/obj/item/gun/ballistic/proc/guncleaning(mob/user, /obj/item/A)
+	if(misfire_probability == 0)
+		to_chat(user, "<span class='notice'>\The [src] seems to be already clean of fouling.</span>")
+		return
+	
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.visible_message("<span class='notice'>[user] begins to cleaning \the [src].</span>", "<span class='notice'>You begin to clean the internals of \the [src].</span>")
+	
+	if(do_after(user, 100, target = src))
+		if(misfire_probability > 0)
+			misfire_probability = 0
+			user.visible_message("<span class='notice'>[user] cleans \the [src] of any fouling.</span>", "<span class='notice'>You clean \the [src], removing any fouling, preventing misfire.</span>")
+			return TRUE
+
+/obj/item/gun/ballistic/wrench_act(mob/living/user, obj/item/I)
+	if(!user.is_holding(src))
+		to_chat(user, "<span class='notice'>You need to hold [src] to modify it.</span>")
+		return TRUE
+	
+	if(!can_modify_ammo)
+		return
+	
+	if(bolt_type == BOLT_TYPE_STANDARD)
+		if(get_ammo())	
+			to_chat(user, "<span class='notice'>You can't get at the internals while the gun has a bullet in it!</span>")
+			return
```

Comment:
won't this stop you from accidentally blowing yourself up by trying to modify a loaded gun?

Reply:
This might have been someone I forgot I might have worked around but at the time I was coding this, I found that the way BOLT_TYPE_STANDARD worked, it was possible to leave ammo in the magazine, change the ammo type and then have the previous ammo still in the gun despite being the alternative ammo.

I should and could have just added a guaranteed misfire chance if that happened, but that might be a little lame to be on the receiving end of.

Reply:
Yeah this strictly isn't going to work for these types of bolts. So I think I'll keep it as is.